### PR TITLE
Fix DS update function

### PR DESCRIPTION
**Pull request by sdodsley**
##### SUMMARY
Ensure DS updates work correctly.

Note an update is not idempotent as a password must be parsed for an update and password changes are not idempotent.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
purefb_ds.py

### CODE REVIEW

**Comment by mahendra-maplelabs** [comment]
@sdodsley This fix working great with Management and SMB directory services but not updating NFS Directory Services.

File: collections/ansible_collections/purestorage/flashblade/plugins/modules/purefb_ds.py [hunk 1]
```diff
@@ -234,38 +234,37 @@ def update_ds(module, blade):
...
-                                mod_ds = True
+            if module.params['dstype'] == 'nfs':
+                if module.params['nis_servers']:
+                    if sorted(module.params['nis_servers']) != sorted(ds_now.nfs.nis_servers) or \
+                            module.params['nis_domain'] != ''.join(map(str, ds_now.nfs.nis_domains)):
+                        attr['nfs'] = {'nis_domains': [module.params['nis_domain']],
+                                       'nis_servers': module.params['nis_servers'][0:30]}
+                        mod_ds = True
+            else:
+                if module.params['uri']:
```
Comment (mahendra-maplelabs) [inline collections/ansible_collections/purestorage/flashblade/plugins/modules/purefb_ds.py:245]: Else only allowing SMB and Management Directory services.  It should allow NFS also if not configured for "NIS" .

**Review by mahendra-maplelabs — CHANGES_REQUESTED** [review]

File: collections/ansible_collections/purestorage/flashblade/plugins/modules/purefb_ds.py (hunk 1 above)
Comment (sdodsley) [reply collections/ansible_collections/purestorage/flashblade/plugins/modules/purefb_ds.py:244]: New push should resolve this oversight.

**Comment by mahendra-maplelabs** [comment]
@sdodsley Tested for updates on NFS, SMB and Management Directory services. Now working fine. Thanks

**Pull request merged**